File: elixir/bird-count/lib/bird_count.ex
defmodule BirdCount do
  def today([]), do: nil
  def today([head | _tail]), do: head

  def increment_day_count([]), do: [1]
  def increment_day_count([head | tail]), do: [head + 1 | tail]

  def has_day_without_birds?([]), do: false

  def has_day_without_birds?([head | tail]) do
    if head == 0 do
      true
    else
      has_day_without_birds?(tail)
    end
  end

  def total([]), do: 0
  def total([head | tail]), do: head + total(tail)

  def busy_days([]), do: 0
  def busy_days([head | tail]) when head >= 5, do: 1 + busy_days(tail)
  def busy_days([_head | tail]), do: busy_days(tail)
end

File: ruby/bird-count/bird_count.rb
class BirdCount
  def self.last_week
    [0, 2, 5, 3, 7, 8, 4]
  end

  def initialize(birds_per_day)
    @birds_per_day = birds_per_day
  end

  def yesterday
    @birds_per_day[-2]
  end

  def total
    @birds_per_day.reduce(:+)
  end

  def busy_days
    raise 'Please implement the BirdCount#busy_days method'
  end

  def day_without_birds?
    raise 'Please implement the BirdCount#day_without_birds method'
  end
end
